### DISCUSSION TITLE
How to install private github repository via npm in github actions workflow ci

### DISCUSSION THREAD

**Discussion by ghost** [community/community#25408] [Actions]
<p>I am trying to install npm dependencies within a github workflow ci by running <code>npm install</code>. However i get the following error:</p>
<pre><code>npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/private-org/private-repo.git
npm ERR!
npm ERR! Warning: Permanently added the RSA host key for IP address 'removed' to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
</code></pre>
<p><strong>**ci.yml**</strong></p>
<pre><code>```
name: CI

on:
push:
branches: [master]

jobs:
test:
runs-on: ubuntu-latest
steps:
- uses: actions/checkout@v2
- name: Use Node.js
uses: actions/setup-node@v1
with:
node-version: '12.x'
- run: node --version
- run: npm install
```
</code></pre>
<p><strong>**package.json**</strong></p>
<pre><code>...
"dependencies": {
"some-pacakage": "git+ssh://git@github.com/private-org/private-repo.gitt",
},
...
</code></pre>
<p>This <code>some-package</code> is being installed via github by npm. The repo is within the same organization as which the workflow is running in. To solve this problem locally you setup ssh key on your github account tied to the organization.</p>
<p>But how can i solve this issue so that its able to install that package via github repo within the workfow ci where im not using my personal github account.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/kaykhan">@kaykhan</a> , </p>
<p>There is <a href="https://github.com/webfactory/ssh-agent" rel="nofollow noopener">an action</a> which could help you setup ssh-agent on GitHub Hosted runner. </p>
<pre><code>- uses: webfactory/ssh-agent@v0.2.0
     with:
        ssh-private-key: ${{ secrets.private_ssh }}
</code></pre>
<p>You could copy the content of private ssh key (~/.ssh/id_rsa under the user folder ) in your local machine to a <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository" rel="nofollow noopener">secret</a> variable private_ssh. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0ba63d7482c34f5d3d2b70c2216732503e2bdd83.png" data-download-href="/uploads/short-url/1F3q1W7ws3Bgrh9ETvy8yGDeM6f.png?dl=1" title="id_rsa.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181090791-bf89685f-a418-4c8b-9ed2-1071e0493e5c.png" alt="id_rsa.png" data-base62-sha1="1F3q1W7ws3Bgrh9ETvy8yGDeM6f" width="468" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0ba63d7482c34f5d3d2b70c2216732503e2bdd83_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">id_rsa.png</span><span class="informations">605×645 9.77 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Then you could use the secret variable in your workflow with syntax ${{ secrets.varname }}. Please add webfactory/ssh-agent action in the front of npm install step. </p>
